<launch>

  <!--  <master auto="start"/>  -->

  <include file="$(find turtlebot_bringup)/launch/minimal.launch"></include>

  <!-- Run the map server 
  <node name="map_server" pkg="map_server" type="map_server" args="$(find my_map_package)/my_map.pgm my_map_resolution"/> --> 
  
  <!-- Map server -->
  <arg name="map_file" default="$(env TURTLEBOT_MAP_FILE)"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />


  <!-- 3D sensor -->
  <arg name="3d_sensor" default="$(env TURTLEBOT_3D_SENSOR)"/>  <!-- r200, kinect, asus_xtion_pro -->

  <include file="$(find turtlebot_bringup)/launch/3dsensor.launch">
    <arg name="rgb_processing" value="false" />
    <arg name="depth_registration" value="false" />
    <arg name="depth_processing" value="false" />

    <!-- We must specify an absolute topic name because if not it will be prefixed by "$(arg camera)".
        Probably is a bug in the nodelet manager: https://github.com/ros/nodelet_core/issues/7 -->
    <arg name="scan_topic" value="/scan" />
  </include>

  <!--- Run AMCL 
  <include file="$(find amcl)/examples/amcl_omni.launch" /> --> 
  
  <!-- AMCL -->
  <arg name="custom_amcl_launch_file" default="$(find turtlebot_navigation)/launch/includes/amcl/$(arg 3d_sensor)_amcl.launch.xml"/>
  <arg name="initial_pose_x" default="0.0"/> <!-- Use 17.0 for willow's map in simulation -->
  <arg name="initial_pose_y" default="0.0"/> <!-- Use 17.0 for willow's map in simulation -->
  <arg name="initial_pose_a" default="0.0"/>
  <include file="$(arg custom_amcl_launch_file)">
    <arg name="initial_pose_x" value="$(arg initial_pose_x)"/>
    <arg name="initial_pose_y" value="$(arg initial_pose_y)"/>
    <arg name="initial_pose_a" value="$(arg initial_pose_a)"/>
  </include>


 <!--- 4/5 are in $(find turtlebot_navigation)/laser/move_base_turtlebot_laser.launch" --> 
<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find argus_move)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find argus_move)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find argus_move)/config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find argus_move)/config/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find argus_move)/config/base_local_planner_params.yaml" command="load" />
</node>

  <node pkg="argus_move" type="map_navigation" name="map_navigation" output="screen">
     <!--<remap from="/scan_testing/cmd_vel" to="keyop_vel_smoother/raw_cmd_vel"/> -->
  </node>

</launch>
